engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbIncidentState.java

```diff
@@ -32,14 +33,14 @@
   private final ColumnFamily<DbLong, Incident> incidentColumnFamily;
 
   /** element instance key -> incident key */
-  private final DbLong elementInstanceKey;
+  private final DbForeignKey<DbLong> elementInstanceKey;
```

Comment:
:thought_balloon: We could think of a naming convention for variables to indicate foreign key, e.g. `elementInstanceFKey` or `fkElementInstance`. I think this would help in the rest of the code to know when `inner()` needs to be called and when not

Reply:
I like `fkElementInstance` but I'm not sure it's really necessary. At least for me it's quite clear that this is a foreign key, just based on inlay hints etc. I'm happy to revisit this though, finding and renaming all foreign keys should be very easy soon :)

Reply:
Sure, I think if we wanted to rename it, we should discuss it in the team. (because then we could also rename composite keys). Just a thought, though